File: CarCareHub_/CarCareHub.Model/ChatKlijentZaposlenik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class ChatKlijentZaposlenik
    {
        public int ChatKlijentZaposlenikId { get; set; }
        public int KlijentId { get; set; }
        public virtual Klijent Klijent { get; set; }
        public virtual Zaposlenik Zaposlenik { get; set; }
        public int ZaposlenikId { get; set; }
        public string Poruka { get; set; }
        public bool PoslanoOdKlijenta { get; set; }
        public DateTime VrijemeSlanja { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_mobile/lib/models/chatKlijentZaposlenik.dart
import 'package:flutter_mobile/models/zaposlenik.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:json_annotation/json_annotation.dart';
part 'chatKlijentZaposlenik.g.dart';

@JsonSerializable()
class chatKlijentZaposlenik {
  int id;
  int klijentId;
  String klijentIme;
  int zaposlenikId;
  String zaposlenikIme;
  String? poruka;
  bool poslanoOdKlijenta;
  DateTime vrijemeSlanja;
//bool? vidljivo;

  chatKlijentZaposlenik(
      this.id,
      this.klijentId,
      this.zaposlenikId,
      this.zaposlenikIme,
      this.klijentIme,
      this.poruka,
      this.poslanoOdKlijenta,
      this.vrijemeSlanja);

  factory chatKlijentZaposlenik.fromJson(Map<String, dynamic> json) =>
      _$chatKlijentZaposlenikFromJson(json);

  Map<String, dynamic> toJson() => _$chatKlijentZaposlenikToJson(this);
}
